packages/core/strapi/lib/services/server/policy.js

```diff
@@ -11,11 +11,22 @@ const resolvePolicies = route => {
   const { pluginName, apiName } = route.info || {};
   const policiesConfig = getPoliciesConfig(route);
 
-  const policies = policiesConfig.map(policyName => {
-    return policy.get(policyName, { pluginName, apiName });
-  });
+  const policiesMiddleware = async (ctx, next) => {
+    const context = policy.createPolicyContext('koa', ctx);
 
-  return [...policies, bodyPolicy];
+    for (const policyName of policiesConfig) {
+      const resolvedPolicy = await policy.get(policyName, { pluginName, apiName });
+      const result = resolvedPolicy({ ctx: context, strapi });
```

Comment:
the policy could be async here
```suggestion
      const result = await resolvedPolicy({ ctx: context, strapi });
```